tests/perf/http_client_perf.cc

```diff
@@ -0,0 +1,150 @@
+            _out = _cs.output();
+            return run_serve_loop();
+        });
+    }
+};
+
+class loopback_http_factory : public http::experimental::connection_factory {
+    loopback_socket_impl lsi;
+public:
+    explicit loopback_http_factory(loopback_connection_factory& f) : lsi(f) {}
+    virtual future<connected_socket> make(abort_source* as) override {
+        return lsi.connect(socket_address(ipv4_addr()), socket_address(ipv4_addr()));
+    }
+};
+
+class client {
+    seastar::http::experimental::client _cln;
+    const unsigned _warmup_limit;
+    const unsigned _limit;
+
+    future<> make_requests(unsigned nr) {
+        for (unsigned i = 0; i < nr; i++) {
+            auto req = http::request::make("GET", "test", "/test");
+            co_await _cln.make_request(std::move(req), [] (const http::reply& rep, input_stream<char>&& in) {
+                return make_ready_future<>();
+            }, http::reply::status_type::ok);
+        }
+    }
+public:
+    client(loopback_connection_factory& lcf, unsigned ops, unsigned warmup)
+            : _cln(std::make_unique<loopback_http_factory>(lcf))
+            , _warmup_limit(warmup)
+            , _limit(ops)
+    {}
+    future<> work() {
+        fmt::print("Warming up with {} requests\n", _warmup_limit);
+        return make_requests(_warmup_limit).then([this] {
+            fmt::print("Warmup finished, making {} requests\n", _limit);
+            auto start = std::chrono::steady_clock::now();
+            auto mallocs = memory::stats().mallocs();
+            return make_requests(_limit).then([this, start, mallocs] {
+                auto delta = std::chrono::steady_clock::now() - start;
+                auto allocs = memory::stats().mallocs() - mallocs;
+                fmt::print("Made {} requests in {}s, {} mallocs\n", _limit, std::chrono::duration_cast<std::chrono::duration<double>>(delta).count(), allocs);
+            });
+        }).finally([this] {
+            return _cln.close();
+        });
+    }
+};
+
+int main(int ac, char** av) {
+    app_template at;
+    namespace bpo = boost::program_options;
+    at.add_options()
+            ("total-ops", bpo::value<unsigned>()->default_value(1000000), "Total requests to make")
+            ("warmup-ops", bpo::value<unsigned>()->default_value(10000), "Requests to warm up")
+            ;
+    return at.run(ac, av, [&at] {
+        auto total_ops = at.configuration()["total-ops"].as<unsigned>();
+        auto warmup_ops = at.configuration()["warmup-ops"].as<unsigned>();
+        return do_with(loopback_connection_factory(1), [total_ops, warmup_ops] (auto& lcf) {
+            auto sf = do_with(server(lcf), [] (auto& srv) {
+                return srv.serve();
+            });
+            auto cf = do_with(client(lcf, total_ops, warmup_ops), [] (auto& cln) {
+                return cln.work();
+            });
+            return when_all(std::move(sf), std::move(cf)).discard_result();
+        });
+    });
```

Comment:
coroutine or thread?

Would be nicer to report insn/op and allocs/op.

Reply:
Thread looks nicer indeed

Report usec/op and allocs/op

Reply:
Why not insn/op (in addition)? It's more stable with todays frequency agile processors.

Reply:
if we don't plan to use `PERF_TEST` because it's complicated to pass command line argument to the individual tests. probably we can still use `linux_perf_event::user_instructions_retired()` directly ?

Reply:
perf-simple-query in scylla